examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -24,16 +27,22 @@ public class WeatherForecastController : ControllerBase
 {
     private static readonly string[] Summaries = new[]
     {
-            "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
     };
 
     private static readonly HttpClient HttpClient = new();
 
     private readonly ILogger<WeatherForecastController> logger;
+    private readonly ActivitySource activitySource;
+    private readonly Counter<long> freezingDaysCounter;
 
-    public WeatherForecastController(ILogger<WeatherForecastController> logger)
+    public WeatherForecastController(ILogger<WeatherForecastController> logger, Instrumentation instrumentation)
     {
         this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+        ArgumentNullException.ThrowIfNull(instrumentation);
+        this.activitySource = instrumentation.ActivitySource;
```

Comment:
Nit: How about we store the Instrumentation object directly in a single field instead of one field per source and counter. It doesn't make a big difference for this sample, but for folks who have many counters it avoids adding lots of fields to their type if they apply the same pattern.

Reply:
I suggested this pattern to simply code elsewhere, but I can see merits to your suggestion too.

this.instrumentation.activitySource.StartActivity, this.instrumentation.counter.Add
vs
this.activitySource.StartActivity, this.counter.Add

Considering this as non-blocking, so proceeding to merge.

Reply:
Are you asking to revert this [suggestion](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4133#discussion_r1098835693)? I will let you and @cijothomas decide.